client/asset/dcr/dcr.go

```diff
@@ -457,100 +464,167 @@ func (dcr *ExchangeWallet) ReturnCoins(unspents asset.Coins) error {
 	}
 	ops := make([]*wire.OutPoint, 0, len(unspents))
 
+	dcr.fundingMtx.Lock()
+	defer dcr.fundingMtx.Unlock()
 	for _, unspent := range unspents {
 		op, err := dcr.convertCoin(unspent)
 		if err != nil {
 			return fmt.Errorf("error converting coin: %v", err)
 		}
 		ops = append(ops, wire.NewOutPoint(&op.txHash, op.vout, op.tree))
+		delete(dcr.fundingCoins, outpointID(&op.txHash, op.vout))
 	}
 	return dcr.node.LockUnspent(true, ops)
 }
 
+// FundingCoins gets funding coins for the coin IDs. The coins are locked. This
+// method might be called to reinitialize an order from data stored externally.
+// This method will only return funding coins, e.g. unspent transaction outputs.
+func (dcr *ExchangeWallet) FundingCoins(ids []dex.Bytes) (asset.Coins, error) {
+	// First check if we have the coins in cache.
+	coins := make(asset.Coins, 0, len(ids))
+	notFound := make(map[string]byte)
```

Comment:
Done. Oddly enough, because of the slight differences in utxo handling, **client/asset/btc** works better with a `struct{}` and **client/asset/dcr** works better with `bool`.